components/RenderProposalContentLambda.tsx

```diff
@@ -0,0 +1,374 @@
+import BigNumber from "bignumber.js";
+import { useState } from "react";
+import { LambdaType, parseLambda } from "../context/parseLambda";
+import { proposalContent } from "../types/display";
+import { crop } from "../utils/strings";
+import { mutezToTez } from "../utils/tez";
+import { walletToken } from "../utils/useWalletTokens";
+import Alias from "./Alias";
+import Tooltip from "./Tooltip";
+
+type data = {
+  label: undefined | string;
+  metadata: undefined | string;
+  amount: undefined | string;
+  addresses: undefined | string[];
+  entrypoints: undefined | string;
+  params: undefined | string;
+};
+
+const RenderProposalContentLambda = ({
+  content,
+  walletTokens,
+}: {
+  content: proposalContent;
+  walletTokens: walletToken[];
+}) => {
+  const [hasParam, setHasParam] = useState(false);
+
+  let data: data = {
+    label: undefined,
+    metadata: undefined,
+    amount: undefined,
+    addresses: undefined,
+    entrypoints: undefined,
+    params: undefined,
+  };
+
+  if ("changeThreshold" in content) {
+    data = {
+      ...data,
+      label: "Update threshold",
+      params: content.changeThreshold.toString(),
+    };
+  } else if ("adjustEffectivePeriod" in content) {
+    data = {
+      ...data,
+      label: "Update proposal duration",
+      params: content.adjustEffectivePeriod.toString(),
+    };
+  } else if ("addOwners" in content) {
+    data = {
+      ...data,
+      label: `Add signer${content.addOwners.length > 1 ? "s" : ""}`,
+      addresses: content.addOwners,
+    };
+  } else if ("removeOwners" in content) {
+    data = {
+      ...data,
+      label: `Remove signer${content.removeOwners.length > 1 ? "s" : ""}`,
+      addresses: content.removeOwners,
+    };
+  } else if ("transfer" in content) {
+    data = {
+      ...data,
+      label: "Transfer",
+      addresses: [content.transfer.destination],
+      amount: `${mutezToTez(content.transfer.amount)} Tez`,
+    };
+  } else if ("execute" in content) {
+    data = {
+      ...data,
+      label: "Execute",
+      metadata: content.execute,
+    };
+  } else if ("executeLambda" in content) {
+    const metadata = JSON.parse(content.executeLambda.metadata ?? "{}");
+
+    const parser = new Parser();
+
+    const [type, lambda] = parseLambda(
+      parser.parseMichelineExpression(content.executeLambda.content ?? "")
+    );
+
+    if (type === LambdaType.LAMBDA_EXECUTION) {
+      data = {
+        ...data,
+        label: "Execute lambda",
+        metadata:
+          metadata.meta === "No meta supplied" ? undefined : metadata.meta,
+        amount: !!lambda?.mutez ? `${mutezToTez(lambda.mutez)} Tez` : undefined,
+        params: metadata.lambda,
+      };
+    } else if (type === LambdaType.CONTRACT_EXECUTION) {
+      data = {
+        ...data,
+        label: "Execute contract",
+        addresses: !!lambda?.contractAddress ? [lambda.contractAddress] : [],
+        entrypoints: lambda?.entrypoint.name,
```

Comment:
if it's `undefined`, the default should be "default"
![image](https://github.com/marigold-dev/tzsafe-ui/assets/115862/564abb6f-0828-43f2-b7d7-1d6f7cb3023b)"